### PR TITLE
feat: added filter by departments for list profiles

### PR DESCRIPTION
**Pull request by rolin999**
### Description

【兼容】list_profile 接口支持 department 过滤

### Checklist

- [ ] 填写 PR 描述及相关 issue (write PR description and related issue)
- [ ] 代码风格检查通过 (code style check passed)
- [ ] PR 中包含单元测试 (include unit test)
- [ ] 单元测试通过 (unit test passed)
- [ ] 本地开发联调环境验证通过 (local development environment verification passed)

### CODE REVIEW

File: src/bk-user/tests/apis/open_v2/test_profiles.py [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+
+    def test_list_with_exist_departments(self, api_client, local_data_source, collaboration_data_source):
+        department_ids = TenantDepartment.objects.values_list("id", flat=True)
+        resp = api_client.get(
+            reverse("open_v2.list_profiles"),
+            data={"lookup_field": "departments", "exact_lookups": ",".join(map(str, department_ids))},
+        )
+        assert resp.status_code == status.HTTP_200_OK
+        assert resp.data["count"] == 20
+
```
Comment (wklken) [inline src/bk-user/tests/apis/open_v2/test_profiles.py:41]: case 需要覆盖一些提前 return 分支，例如 fuzzy 触发 error_codes.VALIDATION_ERROR

File: src/bk-user/bkuser/apis/open_v2/serializers/profilers.py [hunk 2]
```diff
@@ -77,6 +77,7 @@ class ProfileListInputSLZ(ProfileFieldsSLZ):
             "id",
             "username",
             "display_name",
+            "departments",
```
Comment (nannan00) [inline src/bk-user/bkuser/apis/open_v2/serializers/profilers.py:80]: Nit: 关联关系一般放在最后，参考 ProfileFieldsSLZ

File: src/bk-user/bkuser/apis/open_v2/views/profilers.py [hunk 3]
```diff
@@ -380,6 +385,30 @@ def _convert_lookup_field(lookup_field: str, is_exact: bool = True) -> str:
 
         return lookup_field_map[lookup_field] if is_exact else f"{lookup_field_map[lookup_field]}__icontains"
 
+    @staticmethod
+    def _convert_departments_lookup_to_query(values: List[str], is_exact: bool) -> Q | None:
```
Comment (nannan00) [inline src/bk-user/bkuser/apis/open_v2/views/profilers.py:389]: _convert_departments_lookup_to_query => _convert_department_lookup_to_query

File: src/bk-user/tests/apis/open_v2/test_profiles.py (hunk 1 above)
Comment (rolin999) [reply src/bk-user/tests/apis/open_v2/test_profiles.py:41]: Done

File: src/bk-user/bkuser/apis/open_v2/serializers/profilers.py (hunk 2 above)
Comment (rolin999) [reply src/bk-user/bkuser/apis/open_v2/serializers/profilers.py]: Done

File: src/bk-user/bkuser/apis/open_v2/views/profilers.py (hunk 3 above)
Comment (rolin999) [reply src/bk-user/bkuser/apis/open_v2/views/profilers.py]: Done

**Review by nannan00 — APPROVED** [review]

**Pull request merged**